python/pydantic_core/core_schema.py

```diff
@@ -1954,7 +1954,7 @@ class NoInfoValidatorFunctionSchema(TypedDict):
 class WithInfoValidatorFunctionSchema(TypedDict, total=False):
     type: Required[Literal['with-info']]
     function: Required[WithInfoValidatorFunction]
-    field_name: str
+    field_name: str  # deprecated
```

Comment:
I think this `TypedDict` is only used for type hinting -- let me know if this is sufficient or if we should force a deprecation warning here as well somehow.